Initial draft of 3.8.0 release notes (#6218)
medic/cht-core: release-notes/docs/3.8.0.md

# 3.8.0 Release Notes

- [Known issues](#known-issues)
- [Upgrade notes](#upgrade-notes)
- [Tasks data available for analysis](#tasks-data-available-for-analysis)
- [Create Places by SMS](#create-places-by-sms)
- [Replication improvements](#replication-improvements)
- [Performance](#performance)
- [And more...](#and-more)

## Known Issues

No known issues.

## Upgrade Notes

### Breaking changes

> The changes in task storage ([see below](#tasks-data-vailable-for-analysis)) require configuration to be re-deployed using medic-conf 3.1.0 or greater to work properly. Medic-conf continues to work with cht-core releases prior to 3.8.0, so this re-deployment can happen at any time before upgrading.

Reviewer: ```suggestion
The changes in task storage ([see below](#tasks-data-vailable-for-analysis)) require configuration to be redeployed using medic-conf 3.1.0 or greater to work properly. It is recommended to redeploy the configuration before upgrading so your application will continue to work after the upgrade is complete.
```

# 3.8.0 Release Notes

- [Known issues](#known-issues)
- [Upgrade notes](#upgrade-notes)

> - [Tasks data available for analysis](#tasks-data-available-for-analysis)

Reviewer: ```suggestion
- [Task and target data available for analysis](#task-and-target-data-available-for-analysis)
```

- [Create Places by SMS](#create-places-by-sms)
- [Replication improvements](#replication-improvements)
- [Performance](#performance)
- [And more...](#and-more)

> ## Known Issues

Reviewer: ```suggestion
## Known issues
```

Use Sentence Case Everywhere.

No known issues.

> ## Upgrade Notes

Reviewer: ```suggestion
## Upgrade notes
```

### Breaking changes

> The changes to [tasks and targets](#task-and-target-data-available-for-analysis) require a "3.8 compatible" partner configurations to be deployed for the tasks and target tabs to continue to function. You'll see errors on these tabs and the console error `Rules Engine: Updates to the nools schema are required` if the configuration is not cmpatible with 3.8. All "3.8 compatible" configurations are backward compatible and can be deployed safely to any Core Framework version. It is therefore recommended to deploy the "3.8 compatible" configuration before upgrading so your application will work seamlessly once the upgrade completes.

Reviewer: ```suggestion
The changes to [tasks and targets](#task-and-target-data-available-for-analysis) require a "3.8 compatible" partner configurations to be deployed for the tasks and target tabs to continue to function. You'll see errors on these tabs and the console error `Rules Engine: Updates to the nools schema are required` if the configuration is not compatible with 3.8. All "3.8 compatible" configurations are backward compatible and can be deployed safely to any Core Framework version. It is therefore recommended to deploy the "3.8 compatible" configuration before upgrading so your application will work seamlessly once the upgrade completes.
```

Reviewer: Change "cmpatible" to "compatible".

  * For partners using declarative configuration, ensure your configuration is deployed using [medic-conf 3.1.0](https://github.com/medic/medic-conf/blob/master/release-notes.md#31) or greater.
  * For partners maintaining their configuration in the legacy `rules.nools.js` file, ensure the latest [schema changes](https://github.com/medic/medic-projects/commit/3bb4cd5ade155cb7eca0e0ea6791cc82796d3949) are deployed.

These changes also result in new constraints for partner application task and target code:

* Tasks with multiple `actions` cannot use different contacts for their `actions[n].content.contact` values. For example, one action cannot open a form for contact `x` and a second action for the same task open a form for contact `y`.
* Reports "with errors" are now available in all task and target calculations (instead of just some calculations) and partner code is responsible for filtering them as necessary.

> * For partners maintaining their configuration in a `rules.nools.js` file. Multiple contacts emitting tasks with conflicting `id` values is no longer supported. This will result in errant behavior in 3.8.0.

Reviewer: ```suggestion
* For partners maintaining their configuration in a `rules.nools.js` file, multiple contacts emitting tasks with conflicting `id` values is no longer supported. This will result in errant behavior in 3.8.0.
```

* Target emissions with conflicting `id`s were previously merged based on the order in which the requesting contact was created inside CouchDB. It is now merged based on the order of the requesting contact's `reported_date` attribute.

> * The specific meaning of task "resolution" has been augmented to accomodate the additional complexities of reporting a task's "state" in Postgres. (Details here)[https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#core-framework-380-and-later].

Reviewer: ```suggestion
* The specific meaning of task "resolution" has been augmented to accomodate the additional complexities of reporting a task's "state" in Postgres. Read [the documentation](https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#core-framework-380-and-later) for more information.
```

Reviewer: Wording aside, the brackets were in the wrong order so the link formatting didn't work.

> ### High Sync Times Immediately After Upgrade 

Reviewer: ```suggestion
### High sync times immediately after upgrade 
```

> The changes to [tasks and targets](#task-and-target-data-available-for-analysis) may have significant [performance implications](#performance) for some users. Heavy users may see hundreds of new task documents created on their device the first time they load the task or target tabs after upgrading to 3.8.0. Depending on connection fidelity, users can expect sync times for their first sync after viewing the task or target tab in 3.8.0.

Reviewer: ```suggestion
The changes to [tasks and targets](#task-and-target-data-available-for-analysis) may have significant [performance implications](#performance) for some users. Heavy users may see hundreds of new task documents created on their device the first time they load the task or target tabs after upgrading to 3.8.0. Users can expect longer than usual sync times for their first sync after viewing the task or target tab in 3.8.0.
```
